lm_eval/evaluator.py

```diff
@@ -481,18 +466,37 @@ def evaluate(
                     results[task_name][metric + "_stderr" + "," + key] = stderr(items)
 
         if bool(results):
-            for task_or_group in results.keys():
-                for metric in results[task_or_group].keys():
-                    if type(results[task_or_group][metric]) == list:
-                        if "(sample agg)" in metric:
-                            results[task_or_group][metric] = sample_agg_fn[
-                                task_or_group
-                            ][metric](results[task_or_group][metric])
+
+            for group, task_list in reversed(task_hierarchy.items()):
+                versions[group] = "N/A"
+                task_score_dict = {}
+                total_size = 0
+                for task in task_list:
+                    metrics = results[task]
+
+                    current_size = metrics.pop("samples")
+                    # if "size" in metrics:
+                    #     current_size = metrics.pop("size")
+                    # else:
+                    #     current_size = 1
+
+                    for metric in [key for key in metrics.keys() if "_stderr" not in key]:
+
+                        stderr = "_stderr,".join(metric.split(","))
+                        stderr_score = results[task][stderr]
+                        metric_score = results[task][metric]
+
+                        if metric in results[group]:
+                            results[group][metric] = (results[group][metric]*total_size + metric_score*current_size)/(total_size+current_size)
+                            # $$s_z^2 = \frac{(n-1) s_x^2 + (m-1) s_y^2}{n+m-1} + \frac{nm(\bar x - \bar y)^2}{(n+m)(n+m-1)}.$$
+                            results[group][stderr] = ((total_size-1)*results[group][stderr]+(current_size-1)*stderr_score)/(total_size + current_size - 1) \
+                                                        + total_size*current_size/((total_size+current_size)*(total_size+current_size-1))*(results[group][metric] - metric_score)**2
```

Comment:
@StellaAthena added the weighted averaging of subtasks and combining standard error as in 
![image](https://github.com/EleutherAI/lm-evaluation-harness/assets/5774558/5664cacd-e1f8-4ff5-911c-b68b1baf9f7f)

Does the final value need to be square-rooted?

Reply:
Yes, s_x is the standard deviation. s^2_x is the variance.